### PR TITLE
FEAT: add `left_or`, `left_or_else`, `right_or`, `right_or_else` method.

### PR DESCRIPTION
**Pull request by DCjanus**
ref: 

+ https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or
+ https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -384,6 +384,100 @@ impl<L, R> Either<L, R> {
...
+    }
+
+    /// Returns left value or computes it from a closure
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use either::*;
+    /// let left: Either<String, u32> = Left("3".to_string());
+    /// assert_eq!(left.left_or_else(|_|unreachable!()), "3");
```
Comment (cuviper) [inline src/lib.rs]: Style nit:

```suggestion
    /// assert_eq!(left.left_or_else(|_| unreachable!()), "3");
```

File: src/lib.rs [hunk 2]
```diff
@@ -384,6 +384,100 @@ impl<L, R> Either<L, R> {
...
+    ///
+    /// let right: Either<String, u32> = Right(3);
+    /// assert_eq!(right.left_or_else(|x|x.to_string()), "3");
```
Comment (cuviper) [inline src/lib.rs]:
```suggestion
    /// assert_eq!(right.left_or_else(|x| x.to_string()), "3");
```

File: src/lib.rs [hunk 3]
```diff
@@ -384,6 +384,100 @@ impl<L, R> Either<L, R> {
...
+    /// ```
+    pub fn left_or_else<F>(self, f: F) -> L
+        where
+            F: Fn(R) -> L,
```
Comment (cuviper) [inline src/lib.rs]: `FnOnce` is sufficient, and offers the user more flexibility.

```suggestion
            F: FnOnce(R) -> L,
```

File: src/lib.rs [hunk 4]
```diff
@@ -384,6 +384,100 @@ impl<L, R> Either<L, R> {
...
+    }
+
+    /// Returns left value or computes it from a closure
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use either::*;
+    /// let left: Either<String, u32> = Left("3".to_string());
+    /// assert_eq!(left.right_or_else(|x|x.parse().unwrap()), 3);
```
Comment (cuviper) [inline src/lib.rs]:
```suggestion
    /// assert_eq!(left.right_or_else(|x| x.parse().unwrap()), 3);
```

File: src/lib.rs [hunk 5]
```diff
@@ -384,6 +384,100 @@ impl<L, R> Either<L, R> {
...
+    ///
+    /// let right: Either<String, u32> = Right(3);
+    /// assert_eq!(right.right_or_else(|_|unreachable!()), 3);
```
Comment (cuviper) [inline src/lib.rs]:
```suggestion
    /// assert_eq!(right.right_or_else(|_| unreachable!()), 3);
```

File: src/lib.rs [hunk 6]
```diff
@@ -384,6 +384,100 @@ impl<L, R> Either<L, R> {
...
+    /// ```
+    pub fn right_or_else<F>(self, f: F) -> R
+        where
+            F: Fn(L) -> R,
```
Comment (cuviper) [inline src/lib.rs]:
```suggestion
            F: FnOnce(L) -> R,
```

**Comment by DCjanus** [comment]
Sounds good to me. It would be better to be 'rustfmt.toml' in repo root.

**Comment by cuviper** [comment]
Does rustfmt handle code in docs now? That's still listed in its [limitations](https://github.com/rust-lang/rustfmt#limitations).

**Comment by DCjanus** [comment]
`left_or_default` and `right_or_default` has been added.

Thanks for your suggestions. @cuviper

**Comment by cuviper** [comment]
Thanks! I have published 1.5.2.

**Pull request merged**